### PR TITLE
fix(tests): get subset of integration tests passing

### PR DESCRIPTION
**Pull request by mboersma**
This gets four additional cases passing and disables some problematic tests for now. All enabled tests should be expected to pass after this, so we can build incrementally on this base and ensure things stay green. :green_heart:

### CODE REVIEW

File: tests/apps_test.go [hunk 1]
```diff
@@ -101,30 +102,33 @@ var _ = Describe("Apps", func() {
...
 		BeforeEach(func() {
+			os.Chdir("example-go")
 			appName = getRandAppName()
 			cmd := createApp(appName)
 			Eventually(cmd).Should(SatisfyAll(
 				Say("Git remote deis added"),
 				Say("remote available at ")))
-
+			Eventually(cmd).Should(Exit(0))
 			cmd, err := start("GIT_SSH=%s git push deis master", gitSSH)
```
Comment (arschles) [inline tests/apps_test.go]: Worth setting `cmd.Dir = "example-go"` instead of doing the `Chdir`? If so, not blocking for this PR, just a comment.

**Comment by sgoings** [comment]
We expecting people to handle known hosts in the tests?

```
------------------------------
Apps with a deployed app
  can't create an existing app
  /Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:128
Logged in as test-81
Creating Application... done, created test-805056191
Git remote deis added
remote available at ssh://git@deis.199.223.236.30.xip.io:2222/test-805056191.git
The authenticity of host '[deis.199.223.236.30.xip.io]:2222 ([199.223.236.30]:2222)' can't be established.
RSA key fingerprint is d4:1a:c0:76:e8:0f:3b:3d:da:c7:d3:aa:40:c9:b0:4e.
Are you sure you want to continue connecting (yes/no)?
```

**Comment by sgoings** [comment]
I had a few hiccups when running tests against the cluster:

1. see host key acceptance prompt above
2. public key failures:

```
-----------------------------
Apps with a deployed app
  can get app info
  /Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:140
Logged in as test-81
Creating Application... done, created test-875748990
Git remote deis added
remote available at ssh://git@deis.199.223.236.30.xip.io:2222/test-875748990.git
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Destroying test-875748990...
done in 0s

• Failure in Spec Setup (BeforeEach) [2.362 seconds]
Apps
/Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:179
  with a deployed app
  /Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:178
    can get app info [BeforeEach]
    /Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:140

    No future change is possible.  Bailing out early after 0.481s.
    Got stuck at:
        Permission denied (publickey).
        fatal: Could not read from remote repository.

        Please make sure you have the correct access rights
        and the repository exists.

    Waiting for:
        Done, test-875748990:v2 deployed to Deis

    /Users/sethgoings/go/src/github.com/deis/workflow-e2e/tests/apps_test.go:114
```
3. all successful!
4. all successful!
5. failures == apps:with a deployed app:can't create an existing app + apps: with a deployed app: can get app info
6. all successful!
7. all successful!
8. failures == apps:with a deployed app:can't create an existing app + apps: with a deployed app: can get app info

How can we solve the two sporadic "with a deployed app" tests [can't create existing app, can get app info]?

**Comment by mboersma** [comment]
> We expecting people to handle known hosts in the tests?

No--I guess that means the `GIT_SSH` script needs to be updated to avoid known host checking. I'll fix that.

**Comment by mboersma** [comment]
This works well with `make test-integration` outside a cluster, but now times out on the two test cases mentioned by @sgoings when run through the deis-tests helm chart. I think the problem with running this inside a container has to do with the tests still needing to go through the router rather than `DEIS_WORKFLOW_SERVICE_*`. I'll fix that next.

**Pull request merged**